docs/proposals/apparmor.md

```diff
@@ -0,0 +1,242 @@
+are met, the container runtime will configure the appropriate options to apply the profile. Profile
+requirements and defaults can be specified on the
+[PodSecurityPolicy](security-context-constraints.md).
+
+## Prerequisites
+
+When an AppArmor profile is specified, the Kubelet will verify the prerequisites for applying the
+profile to the container. In order to [fail
+securely](https://www.owasp.org/index.php/Fail_securely), a container **will not be run** if any of
+the prerequisites are not met. The prerequisites are:
+
+1. **Kernel support** - The AppArmor kernel module is loaded. Can be checked by
+   [libcontainer](https://github.com/opencontainers/runc/blob/4dedd0939638fc27a609de1cb37e0666b3cf2079/libcontainer/apparmor/apparmor.go#L17).
+2. **Runtime support** - For alpha, Docker will be required (rkt does not currently have AppArmor
+   support). All supported Docker versions include AppArmor support. See
+   [Container Runtime Interface](#container-runtime-interface) for other runtimes.
+3. **Installed profile** - The target profile must be loaded prior to starting the container. Loaded
+   profiles can be found in the AppArmor securityfs \[1\].
+
+If any of the prerequisites are not met an event will be generated to report the error and the pod
+will be
+[rejected](https://github.com/kubernetes/kubernetes/blob/cdfe7b7b42373317ecd83eb195a683e35db0d569/pkg/kubelet/kubelet.go#L2201)
+by the Kubelet.
+
+*[1] The securityfs can be found in `/proc/mounts`, and defaults to `/sys/kernel/security` on my
+Ubuntu system. The profiles can be found at `{securityfs}/apparmor/profiles`
+([example](http://bazaar.launchpad.net/~apparmor-dev/apparmor/master/view/head:/utils/aa-status#L137)).*
+
+## API Changes
+
+The intial alpha support of AppArmor will follow the pattern
+[used by seccomp](https://github.com/kubernetes/kubernetes/pull/25324) and specify profiles through
+annotations. Profiles can be specified through pod annotations, as either a container
+profile, a pod profile (applied to all pod containers), or both (in which case the container
+annotation overrides the pod annotation). The annotation format is a URI key, and a profile name
+value:
+
+- `apparmor.security.alpha.kubernetes.io/container/<container_name>`
+- `apparmor.security.alpha.kubernetes.io/pod`
+
+The profiles can be specified in the following formats (following the convention used by [seccomp](../../docs/design/seccomp.md#api-changes)):
+
+1. `runtime/default` - Applies the default profile for the runtime. For docker, the profile is
+   generated from a template
+   [here](https://github.com/docker/docker/blob/master/profiles/apparmor/template.go). If no
+   AppArmor annotations are provided, this profile is enabled by default if AppArmor is enabled in
+   the kernel.
+2. `localhost/<profile_name>` - The profile name specifies the profile to load.
+
+*Note: There is no way to specify an "unconfined" profile, since it is discouraged. If this is truly
```

Comment:
also if the container is privileged might want to note that "docker" will not use apparmor unless a custom profile name is passed